docs/plugin-development-quickstart.md

```diff
@@ -4,28 +4,185 @@ author: Randolph
 
 # Quick start for Plugin Development
 
-## 1. Start a plugin file
+## 1. Introduction
 
-input following code to start a plugin file
+This is an example of how to develop plugin according to tools provided by Mask. And our task is to develop an app related to Ethereum Name Service, people can purchase ENS in application board and see details about specific ENS domain in timeline and profile of twitter.
+
+## 2. Start a plugin file
+
+You can create a plugin folder by entering following command
 
 > npx gulp new-pkg
 
-...(a brief introduction about the files generated)
+After entering related information about the plugin you want to create, a plugin folder will be created in `packages/plugins`. This folder contains lots of files, and core files are `SNSAdaptor` and `Worker`. The former contains all the code related to front-end interaction, the latter plays a database-like role.
+Since we want to add a application into maskbook, we need to add related config:
+
+`pnpm-lock.yaml`
+
+```JavaScript
+    '@masknet/plugin-ENS': workspace:*
+    '@masknet/plugin-ENS': link:../plugins/ENS
+```
+
+`packages/mask/package.json`
+
+```JavaScript
+    "@masknet/plugin-ENS": "workspace:*",
+```
+
+`packages/mask/.webpack/config.ts`
+
+```JavaScript
+    '@masknet/plugin-ENS': join(__dirname, '../../plugins/your file name/src/'),
+```
+
+`packages/mask/src/plugin-infra/register.js`
```

Comment:
`.js`?

Reply:
Yes. This is a js file for some TypeScript reason